src/engine/condition.ts

```diff
@@ -56,3 +62,63 @@ const createCondition = (condition: EngineCondition) => (
 };
 
 export default createCondition;
+
+/* Matching conditions pieces */
+
+const isArrayIntersects =
+  (features: PageFeatureResult, query: EngineConditionQuery<ArrayIntersectsFilter>): boolean =>
+  !!features &&
+  versionMatches(features, query) &&
+  stringArrayFilterMatches(filters.arrayIntersects, features, query);
+
+const isVectorDistanceLesserThanThreshold =
```

Comment:
I am surprised lint has not picked up on this... but this should be declared prior to use.

Reply:
I guess `no-use-before-define` is not enabled by the [recommended settings](https://eslint.org/docs/rules/).